### PR TITLE
Add qdht::Id class

### PR DESCRIPTION
**Pull request by fritzm**
This PR introduces the basic qdht::Id class (node and message IDs for qdht) and associated unit tests.  A skeleton README.md for qdht was also started here.

### CODE REVIEW

File: qdht/Id.cc [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+using namespace std;
+
+namespace lsst {
+namespace qserv {
+namespace qdht {
+
+
+Id::Id()
+:   msb{0}, xsb{0}, lsb{0}
```
Comment (jgates108) [inline qdht/Id.cc:20]: Why not define default values with the variable definitions in the header file?

Comment (fritzm) [inline qdht/Id.cc:20]: Just not used to doing that yet, ha!  Will do.

File: qdht/testqdht.cc [hunk 2]
```diff
@@ -1,10 +1,243 @@
...
+BOOST_AUTO_TEST_CASE(id_serdes)
+{
+    qdht::Id a, b;
+    a.parse("00112233445566778899AABBCCDDEEFF12345678");
+    uint8_t cereal[20];
+    a.serialize(cereal);
+    b.deserialize(cereal);
+    BOOST_TEST(string(b) == "00112233445566778899AABBCCDDEEFF12345678");
+}
+
```
Comment (jgates108) [inline qdht/testqdht.cc:241]: Nice collection of unit test.

File: qdht/Id.cc [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+}
+
+
+bool Id::operator<(Id const& rhs) const
+{
+    return (msb == rhs.msb)
+        ? (xsb == rhs.xsb)
+            ? (lsb < rhs.lsb)
+            : (xsb < rhs.xsb)
+        : (msb < rhs.msb);
```
Comment (jgates108) [inline qdht/Id.cc:65]: This isn't easy to read.

**Pull request closed without merging**